### PR TITLE
Convert HTML to PDF with Docraptor

### LINKED ISSUE #4179: [Languages] Convert HTML to PDF with DocRaptor
#### Proposed title of article
Convert HTML to PDF with DocRaptor

#### Proposed article introduction
Producing PDFs is a typical programming task. PDFs are utilized to make solicitations, receipts, eBooks, distributing reports and the sky is the limit from there. Tragically, automatic age of PDF can be troublesome, particularly in the event that you need to create numerous PDFs quickly. 
DocRaptor is a HTML-to-PDF API that extraordinarily improves on the issue, and they have a PHP library for simple joining. In this instructional exercise, learners will be able to  stroll through how to change over HTML into a PDF with DocRaptor:
#### Key takeaways 
After reading this article learners will be conversant in:
- Installation
- Authentication
- Add HTML or a URL
- API Options
- Get the document!

#### Article quality
This article will help learners with the basics they need to get started with DocRaptor. Many articles out there  only rely on official documentation and some of the examples they offer are a bit too harsh for novices in PHP. This article will be ideal for novices and the code snippets involved will be illustrated clearly and will be straight on point.

#### References
N/A

### PR DESCRIPTION
**Pull request by taves-hub**
#### Formatting and structure checks
- [X] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [X] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [X] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [X] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [X] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [X] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [X] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [X] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [X] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images    
#### Grammar & plagiarism checks
- [X] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [X] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [X] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?
#### Technical checks
- [X] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [X] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [X] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?
#### Topic suggestion this closes
This closes #4179

### CODE REVIEW

**Comment by taves-hub** [comment]
@mercymeave All set

File: content/articles/convert-html-to-pdf-with-docraptor/index.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+- [API options](#api-options)
+- [Get documentation](#get-documentation)
+- [Conclusion](#conclusion)
+  - [Further activity reading](#further-activity-reading)
+  
+### Installation
+Download the most recent version of DocRaptor to your computer and unzip it into your hard drive. You have two options for for doing this the first is using an author and the second is when working alone.
+
+If you are collaborating with an author, you will need to mention their name in the purchase line of your order confirmation.
+
```
Comment (mercymeave) [inline content/articles/convert-html-to-pdf-with-docraptor/index.md:19]: Elaborate on how this is done clearly. I am trying to understand where the author is coming from. Let the reader know the complete steps of installation. Remember we are writing for beginners as well.

File: content/articles/convert-html-to-pdf-with-docraptor/index.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+```javascript
+composer require docraptor/docraptor
+```
+
+- If you are not working with an author, you should download the most recent version and unzip it into your work area instead. Then, create a new record with the name `docraptor.php` and include the file `autoload.php` in it.
+
+```javascript
+require_once('./path/to/docraptor-php/autoload.php');
+```
```
Comment (mercymeave) [inline content/articles/convert-html-to-pdf-with-docraptor/index.md:28]: Is this Javascript or php?

File: content/articles/convert-html-to-pdf-with-docraptor/index.md [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+### Authentication
+You can use the API key `KEY YOUR API HERE`, which will not require you to have a record, but this will only work with watermarked archives and will not work with unwatermarked archives. In any case, you will need to have a track record that is up to date.
+
+```javascript
+$configuration=DocRaptor\configuration::getDefaultconfiguration();
+$configuration->setUsername('KEY_YOUR_API_HERE');
+```
```
Comment (mercymeave) [inline content/articles/convert-html-to-pdf-with-docraptor/index.md:36]: What are archives in this case? Would someone who has no programming knowledge know what you mean here?

File: content/articles/convert-html-to-pdf-with-docraptor/index.md [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+```javascript
+$docraptor = newDocRaptor\DocApi();
+$doc = newDocRaptor\doc();
+$doc = setDocumentUrl("http: //DocRaptor.com/returns/sales.html");
+```
+
+Whether you use HTML or URLs in your report, all of the resources you reference should be accessible via the internet for DocRaptor to work.
+
+### API options
+Given that DocRaptor may also convert reports into various records, such as HTML to Excel, we must create the right archive for each report.
```
Comment (mercymeave) [inline content/articles/convert-html-to-pdf-with-docraptor/index.md:58]: How does one convert a report into a record?

File: content/articles/convert-html-to-pdf-with-docraptor/index.md [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+```javascript
+try{
+    $creat-response=$DocRaptor->creatDoc($doc);
+} catch (DocRaptor\ApiException $Error){
+    echo $error ."\n";
+    echo $error->getMessage() ."\n";
+    echo $error->getCode() ."\n";
+    echo $error->getResponseBody() ."\n";
+}
+```
```
Comment (mercymeave) [inline content/articles/convert-html-to-pdf-with-docraptor/index.md:89]: You wrote a PHP script and identified it as javascript.

**Review by mercymeave — CHANGES_REQUESTED** [review]
Hello @taves-hub :wave: 

Thank you for the article. You have a great article here that will benefit most readers in working with pdfs and HTML.

However, there are a few concerns we need to address.

- Give a  clear explanation of how one will set up and use the API.
- For this case, it seems like you are working on a project. Provide details of the project from prerequisites, software to work with, initialization, folder structure and files to work with.
- In every step, mention the file you are working on and use the appropriate snippet identifier.
- Add proper comments and explanations to your code.
- Finally provide a link to the project in github.

I also suggest you go through the docs to see the structure and in-depth format of the articles.
Work on these changes so that we proceed. 

I will be closing this PR  to create more room as I wait for you to work on the changes.  It is not a problem just ping me to reopen it after you are done working on the changes.

**Comment by taves-hub** [comment]
@mercymeave  I'm done with all the changes that were required to move, everything is set.

**Comment by mercymeave** [comment]
I will re-open the PR then you push the changes.

**Comment by taves-hub** [comment]
@mercymeave, requested changes have been executed accordingly

**Comment by mercymeave** [comment]
Hello @taves-hub,

I have gone through the changes. You have a great piece of article that would help our readers. However apart from what is in the documentation, would you please provide real-life use cases? 

The example you used is similar to what they used in the documentation.

Think of something like how to create a website and include it as a plugin that converts a specific webpage to a pdf. Take for instance I have a website that displays records or generates receipts, how would I implement Docraptor to perform this action for my site?

I think giving readers this level of content will yield more value to our platform and your portfolio as well.

No rush just take your time. Ping me once you are done.

cc: @hectorkambow

**Review by mercymeave — CHANGES_REQUESTED** [review]
Hello @taves-hub,

I have gone through the changes. You have a great piece of article that would help our readers. However apart from what is in the documentation, would you please provide real-life use cases? 

The example you used is similar to what they used in the documentation.

Think of something like how to create a website and include it as a plugin that converts a specific webpage to a pdf. Take for instance I have a website that displays records or generates receipts, how would I implement Docraptor to perform this action for my site?

I think giving readers this level of content will yield more value to our platform and your portfolio as well.

No rush just take your time. Ping me once you are done.

cc: @hectorkambow

**Comment by hectorkambow** [comment]
👍 Great feedback and guidance @mercymeave 

🚀  

Well done - if more time is needed - we can close the PR to allow the author the proper time to make the required edits.

**Comment by kymomolly** [comment]
@mercymeave working on it

**Comment by taves-hub** [comment]
@mercymeave  working on it

**Comment by taves-hub** [comment]
@mercymeave, requested illustration committed

**Comment by kymomolly** [comment]
Hi @mercymeave what next

**Comment by taves-hub** [comment]
Hi @mercymeave what next?

**Review by mercymeave — CHANGES_REQUESTED** [review]
Hello @taves-hub 

Thank you for adding more content to the article. However, kindly ensure that all content must appear in the `index.md` file. Add it to the index file with a detailed explanation then we proceed.

**Comment by taves-hub** [comment]
@mercymeave, necessary changes have been put into place

**Comment by mercymeave** [comment]
ON IT.

**Comment by hectorkambow** [comment]
@taves-hub  

If any other edits are needed during the tech review - we will be closing this PR and please re-submit when it is ready to be published as a new PR (and you can add the duplicate label) and ping the reviewer.  👍  

cc @mercymeave  

As a rule of thumb - please only submit articles (pull requests) with a complete and fully polished article - ready to be published. 
 
All revisions and edits should be completed on your own forked repo (as to not take up room in the queue). 

Closing a PR does NOT mean the article can not be published (or fixed) it simply means that the edits, revisions, and fixes will happen outside of the queue.
 
 🚀

**Comment by taves-hub** [comment]
@mercymeave ,@hectorkambow, i believe that this article is more than ready to proceed to final review. Thank you

**Comment by jerimkaura** [comment]
> @mercymeave ,@hectorkambow, i believe that this article is more than ready to proceed to final review. Thank you

I urge you to be patient as I work on refining your article. We encourage authors to be patient so that we don't rush publishing article that are not polished enough.

**Comment by mercymeave** [comment]
Hello @taves-hub 

I urge you to be patient as I work on polishing this article. We highly encourage publishing very well polished articles for the benefit of our readers. Thank you for your patience.

**Review by mercymeave — CHANGES_REQUESTED** [review]
Hello @taves-hub 

Kindly remove this folder and its file. `illustration-on-docraptor/converted-pdf-with-docraptor.png ` then the article will be ready for final review.

**Comment by taves-hub** [comment]
@mercymeave, Done

**Comment by mercymeave** [comment]
Hello, you don't have a `hero` image. Secondly, delete one of the redundant `converted-pdf-with-docraptor.png` file. Just one is enough.

**Comment by hectorkambow** [comment]
@mercymeave  

This PR should have been closed days ago

**Comment by hectorkambow** [comment]
@taves-hub Please be sure to go through ALL the edits by our peer reviewer and make the last edit required. 

Place all these in a new PR and we can add a duplicate label to it. 

This PR has gone way far into revision - our team can not provide revisions services. 

Our team can only review before publishing. 👍

**Comment by hectorkambow** [comment]
@mercymeave 

This should NOT occur:
>I urge you to be patient as I work on polishing this article. We highly encourage publishing very well polished articles for the benefit of our readers. Thank you for your patience.

**Comment by hectorkambow** [comment]
> 👍 Great feedback and guidance @mercymeave
> 
> 🚀
> 
> Well done - if more time is needed - we can close the PR to allow the author the proper time to make the required edits.

We should have closed this around 10 days ago - to allow the author the time to properly make the edits - WE ARE NOT TO MAKE THE EDITS FOR THE AUTHORS.

**Pull request closed without merging**